interfaces/builtin/shared_memory.go

```diff
@@ -91,9 +91,10 @@ const sharedMemoryBaseDeclarationSlots = `
         - app
         - gadget
         - core
-      slot-snap-id:
-        - PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4
-        - 99T7MUlRhtI3U0QFgl5mXXESAiSwt776
+    deny-installation:
+      slot-snap-type:
+        - app
+        - gadget
```

Comment:
I recommend to come up with different (non opposite) key names instead of using the `allow-installation` and `deny-installation` pair, because seeing declarations like:
```
    allow-installation:
      slot-snap-type:
        - app
        - gadget
        - core
    deny-installation:
      slot-snap-type:
        - app
        - gadget
```
is going to be confusing no matter how well you document it :-)

Reply:
Introducing new keys is costly across the ecosystems of tools we have. At some point we might be forced to add some base-declaration-only functionality (ATM everything that goes/is supported in the base-declaration and the snap-declaration is the same) which might be a good time to add maybe a base-declaration-only keyword for some of this.